seleniumbase/core/capabilities_parser.py

```diff
@@ -169,11 +171,27 @@ def get_desired_capabilities(cap_file):
         if data:
             key = data.group(1)
             value = False
-            desired_capabilities[key] = value
-            num_capabilities += 1
+            capabilities[key] = value
             continue
 
-    if num_capabilities == 0:
+    return capabilities
+
+
+def get_desired_capabilities(cap_file):
+    f = open(cap_file, 'r')
+    all_code = f.read()
+    f.close()
+
+    if cap_file.endswith('.py'):
+        capabilities = _parse_ast(all_code)
+        if not capabilities:
+            capabilities = _parse_manual(all_code)
+    elif cap_file.endswith('.json'):
+        capabilities = json.loads(all_code)
+    else:
+        raise Exception("\n\n`%s` is not a Python or JSON file!\n" % cap_file)
```

Comment:
This part should happen before ``f = open(cap_file, 'r')`` if the file type is invalid.

Reply:
fixed